docs/mshv.md

```diff
@@ -0,0 +1,52 @@
+# MSHV
+
+The Microsoft Hypervisor is a mature virtualization platform on its own right and Microsoft has great expertise. It is battle tested in Azure and on-premises deployment, and the corner stone of some features in Windows [Device guard and confidential guard](https://techcommunity.microsoft.com/t5/iis-support-blog/windows-10-device-guard-and-credential-guard-demystified/ba-p/376419) and [WSL2](https://docs.microsoft.com/en-us/windows/wsl/wsl2-faq)
+
+Microsoft till now has only run Windows as the root OS. Companies like Amazon, Google and Alibaba have long been using Linux as the primary driver for their cloud offerings. Linux is pervasive on both the host and guest side in their clouds. These companies and other open source developers have made many contributions to the Linux OS for supporting virtualization technology and Linux has become feature rich in this area. By limiting the root OS to Windows, Microsoft is missing out on these innovations that can be leveraged to support a vast array of Microsoft’s business needs, with quick turnaround solutions.
+
+In the recent years, Microsoft has added Linux as Dom-0 support to the Microsoft hypervisor. It open sourced different components as follows
+
+    1. Added support to linux tree to boot on Microsoft Hyper-V
+    2. Linux module to mane VMs (mshv)  that exposes /dev/mshv
+    3. Rust equivalent bindings and IOCTL wrappers
+    4. IGVM related crates
+
+## Components
+
+The following components are related to MSHV support with Cloud-Hypervisor:
+
+* [igvm-crates](https://github.com/microsoft/igvm) : Parsing IGVM file
+
+* [mshv-crates](https://github.com/rust-vmm/mshv) : Rust crates to interact with Kernel Module (/dev/mshv)
+
+* [igvm-tooling](https://github.com/microsoft/igvm-tooling) : Tool to generate IGVM file
+
+## IGVM
+
+Independent Guest Virtual Machine (IGVM) file format.The format specification can be found in the igvm_defs crate, with a Rust implementation of the binary format in the igvm crate.
+
+The IGVM file format is designed to encapsulate all information required to launch a virtual machine on any given virtualization stack, with support for different isolation technologies such as AMD SEV-SNP and Intel TDX.
+
+At a conceptual level, this file format is a set of commands created by the tool that generated the file, used by the loader to construct the initial guest state. The file format also contains measurement information that the underlying platform will use to confirm that the file was loaded correctly and signed by the appropriate authorities.
+
+Cloud Hypervisor can be built using igvm feature flag along with mshv and/or sev-snp
+
+## SEV-SNP
+
+ AMD introduced Secure Encrypted Virtualization (SEV). It also added Platform Security Processor (PSP) firmware
```

Comment:
Rephrase with a little more detail and a link, perhaps? e.g.

"
AMD's [Secure Encrypted Virtualization (SEV)](https://www.amd.com/en/developer/sev.html) and extensions such as Secure Nested Paging (SEV-SNP) encrypt memory and restrict access to a guest VM's memory and registers, securing it against a compromised hypervisor or VMM. They utilize the Platform Security Processor (PSP) to store keys and encrypt/decrypt the data.
"
Then the bit about Microsoft adding support etc...